File: components/dropdown/dropdown-menu.ts
import {Directive, ElementRef, Host, OnInit, Input} from 'angular2/core';
import {Dropdown} from './dropdown';

@Directive({ selector: '[dropdown-menu]' })
export class DropdownMenu implements OnInit {
  @Input() public templateUrl:string;

  constructor(@Host() public dropdown:Dropdown, public el:ElementRef) {
  }

  ngOnInit() {
    this.dropdown.dropDownMenu = this;
  }
}

File: lib/dropdown/dropdown-menu.dart
part of ns_dropdown;

@Directive (selector: "[dropdown-menu], .dropdown-menu",
    inputs: const [ "templateUrl"])
class DropdownMenu implements DropdownMenuInterface, OnInit {
  Dropdown dropdown;

  ElementRef el;

  String templateUrl;

  DropdownMenu(@Host () this .dropdown, this .el) {}

  ngOnInit() {
    dropdown.dropDownMenu = this;
  }
}